File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    dic = {}
    answer = []
    for i in range(len(s)):
        if s[i] not in dic:
            dic[s[i]] = i
            answer.append(-1)
        else:
            v = i - dic[s[i]]
            answer.append(v)
            dic[s[i]] = i
            
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        Map<Character, Integer> dic = new HashMap<>();
        List<Integer> result = new ArrayList<>();
        
        for (int i=0; i<s.length(); i++) {
            Character ctr = s.charAt(i);
            if (dic.containsKey(ctr)) {
                result.add(i - dic.get(ctr));                
            } else {
                result.add(-1);
            }
            dic.put(ctr, i);
        }
        
        return result.stream().mapToInt(v -> v).toArray();
    }
}